sdk/metric/provider.go

```diff
@@ -69,10 +72,16 @@ func NewMeterProvider(options ...Option) *MeterProvider {
 //
 // This method is safe to call concurrently.
 func (mp *MeterProvider) Meter(name string, options ...metric.MeterOption) metric.Meter {
-	// TODO (#2821): ensure this is concurrent safe.
-	// TODO: test this is concurrent safe.
-	// TODO (#2821): register and track the created Meter.
-	return &meter{}
+	c := metric.NewMeterConfig(options...)
+	m, existing := mp.meters.Get(instrumentation.Library{
+		Name:      name,
+		Version:   c.InstrumentationVersion(),
+		SchemaURL: c.SchemaURL(),
+	})
+	if !existing {
+		m.provider = mp
+	}
+	return m
```

Comment:
this cyclomatic complexity would be reduced if we pass the `mp *MeterProvider` to the `meterRegistry.Get()` so that it can assign the meter provider when it is created

Reply:
There will be more to setup in the future here. Having a way to determine if the meter existed already and perform additional setup here is ideal. It prevents scope creeping of the `meterRegistry`'s `Get` method from getting a `meter` from the registry, to getting the meter, initializing it, and any thing else that needs to be done.